citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AsyncCitationProcessor.java

```diff
@@ -119,24 +125,64 @@ public void sync(IUser user, String groupId, long contentVersion, String collect
         jobRepo.save(job);
 
         AtomicInteger counter = new AtomicInteger();
+        
+        if(checkIfThreadIsInterruptedAndCancelJob(job, groupId)) {
+            return new AsyncResult<String>(job.getId());
+        }
+
         syncCitations(user, groupId, job, versions, counter);
+
+        if(checkIfThreadIsInterruptedAndCancelJob(job, groupId)) {
```

Comment:
Actually, now that I'm looking at this again, you will want to cancel the thread within the `syncCitations` and probably also `syncCollections` method. They are the long running methods that need to be cancelled if cancellation was requested. Otherwise, if there are 1000 citations to sync it would still go through all of them before cancelling.